src/extended_json.ts

```diff
@@ -70,16 +72,22 @@ const keysToCodecs = {
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 function deserializeValue(value: any, options: EJSONOptions = {}) {
   if (typeof value === 'number') {
+    const in32BitRange = value <= BSON_INT32_MAX && value >= BSON_INT32_MIN;
+    const in64BitRange = value <= BSON_INT64_MAX && value >= BSON_INT64_MIN;
```

Comment:
We should start a thread with the team, this auto upgrading of an int64 is an outstanding bug with respect to Longs. its questionable whether we want bigints to do the correct thing or mirror the current behavior. (NODE-4377).

Reply:
From the specs:

> If the number is an integer, parsers SHOULD interpret it as being of the smallest BSON integer type that can represent the number exactly.

okay so from the specs, the behavior is correct here, sorry that linked ticket is with respect to something else. And I see the examples from the design doc aligns with this so disregard. When the output would've been a Long we're changing it to bigint, but when outputs would've been an int32 it stays as a number/Int32 depending on relaxed.